### PR TITLE
Update detekt and fix some more ip violations in gradle/gradle

### PR DESCRIPTION
**Pull request by asodja**
Part of https://github.com/gradle/gradle/issues/29374

### CODE REVIEW

**Comment by asodja** [comment]
@bot-gradle test this

**Comment by asodja** [comment]
@bot-gradle test this

File: build-logic/performance-testing/src/main/groovy/gradlebuild.performance-templates.gradle [hunk 1]
```diff
@@ -589,8 +589,8 @@ performanceTest.registerTestProject("workerApiProject", Copy) {
 }
 
 tasks.withType(JvmProjectGeneratorTask).configureEach {
-    if (project.hasProperty("springDmPluginVersion")) {
-        templateArgs['springDmPluginVersion'] = springDmPluginVersion
+    if (providers.gradleProperty("springDmPluginVersion").isPresent()) {
+        templateArgs['springDmPluginVersion'] = providers.gradleProperty("springDmPluginVersion").get()
```
Comment (alllex) [inline build-logic/performance-testing/src/main/groovy/gradlebuild.performance-templates.gradle:593]: 💅  Maybe extract the property into a local variable?

**Review by alllex — APPROVED** [review]
LGTM!

File: build-logic-commons/gradle-plugin/build.gradle.kts [hunk 2]
```diff
@@ -20,5 +20,5 @@ dependencies {
     implementation("org.jetbrains.kotlin:kotlin-gradle-plugin:2.0.21")
     implementation("org.gradle:test-retry-gradle-plugin:1.5.2")
 
-    implementation("io.gitlab.arturbosch.detekt:detekt-gradle-plugin:1.23.6")
+    implementation("io.gitlab.arturbosch.detekt:detekt-gradle-plugin:1.23.7")
```
Comment (alllex) [inline build-logic-commons/gradle-plugin/build.gradle.kts:23]: ❔  Does this remove all Detekt-related IP violations in the build?

File: build-logic/performance-testing/src/main/groovy/gradlebuild.performance-templates.gradle (hunk 1 above)
Comment (asodja) [reply build-logic/performance-testing/src/main/groovy/gradlebuild.performance-templates.gradle]: Done

**Pull request merged**